apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Tests/BigQueryInsertResultsTests.cs

```diff
@@ -0,0 +1,426 @@
+                new BigQueryInsertRowErrors(rows[6], new List<InsertErrorsData> { row6List1 }),
+            };
+            List<SingleError> expectedSingleErrors = (from rowError in expectedInsertRowErrors
+                                                      from detail in rowError
+                                                      select detail).ToList();
+
+            BigQueryInsertResults results = new BigQueryInsertResults(
+                _client,
+                new InsertOptions { SkipInvalidRows = true },
+                rows,
+                response);
+
+            Assert.Equal(BigQueryInsertStatus.SomeRowsInserted, results.Status);
+            Assert.True(results.HasErrors);
+            Assert.Same(results, results.ThrowOnNoneInserted());
+            AssertException(
+                Assert.Throws<GoogleApiException>(() => results.ThrowOnNotAllInserted()));
+            AssertException(
+                Assert.Throws<GoogleApiException>(() => results.ThrowOnAnyError()));
+
+            Assert.Equal(expectedInsertRowErrors, results.Errors.ToList(), new BigQueryInsertRowErrorEqualityComparer());
+
+            void AssertException(GoogleApiException exception)
+            {
+                Assert.Equal("fake_service_name", exception.ServiceName);
+                Assert.Contains(BigQueryInsertStatus.SomeRowsInserted.ToString(), exception.Message);
+                Assert.Equal(expectedSingleErrors, exception.Error.Errors, new SingleErrorEqualityComparer());
+            }
+        }
+
+        [Fact]
+        public void NoRowsInserted_NoSkip()
+        {
+            ErrorProto row1Error1 = new ErrorProto { Location = "field_1", Reason = "reason_1", Message = "message_1" };
+            ErrorProto row1Error2 = new ErrorProto { Location = "field_2", Message = "message_2" };
+            InsertErrorsData row1List1 = new InsertErrorsData { Index = 1, Errors = new List<ErrorProto> { row1Error1, row1Error2 } };
+
+            ErrorProto row5Error1 = new ErrorProto { Location = "field_3", Reason = "reason_3" };
+            InsertErrorsData row5List1 = new InsertErrorsData { Index = 5, Errors = new List<ErrorProto> { row5Error1 } };
+
+            InsertErrorsData row6List1 = new InsertErrorsData { Index = 6 };
+
+            ErrorProto row1Error3 = new ErrorProto { Location = "field_4", Reason = "reason_4", Message = "message_4" };
+            InsertErrorsData row1List2 = new InsertErrorsData { Index = 1, Errors = new List<ErrorProto> { row1Error3 } };
+
+            TableDataInsertAllResponse response = new TableDataInsertAllResponse
+            {
+                InsertErrors = new List<InsertErrorsData>
+                {
+                    row1List1,
+                    row5List1,
+                    row6List1,
+                    row1List2
+                }
+            };
+
+            IList<BigQueryInsertRow> rows = new List<BigQueryInsertRow>
```

Comment:
We might want a helper method for this, rather than taking 11 lines in each test. Or just use `Enumerable.Range(0, 8).Select(_ => new BigQueryInsertRow()).ToList()`